app/celery/tasks.py

```diff
@@ -205,12 +208,17 @@ def send_email(self, service_id,
 
         current_app.logger.info("Email {} created at {}".format(saved_notification.id, created_at))
     except SQLAlchemyError as e:
-        current_app.logger.exception("RETRY: send_email notification".format(notification.get('job', None),
-                                                                             notification.get('row_number', None)), e)
-        try:
-            raise self.retry(queue="retry", exc=e)
-        except self.MaxRetriesExceededError:
-            current_app.logger.error(
-                "RETRY FAILED: task send_email failed for notification".format(
-                    notification.get('job', None),
-                    notification.get('row_number', None)), e)
+        if not get_notification_by_id(notification_id):
+            # Sometimes, SQS plays the same message twice. We should be able to catch an IntegrityError, but it seems
+            # SQLAlchemy is throwing a FlushError. So we check if the notification id already exists then do not
+            # send to the retry queue.
+            current_app.logger.exception("RETRY: send_email notification".format(notification.get('job', None),
+                                                                                 notification.get('row_number', None)),
+                                         e)
```

Comment:
this won't work - the string format will fail and you shouldn't pass in `e` to a `logger.exception` call as it is automatically attached.